### PR TITLE
Array_and_Equilibrium

### PR DESCRIPTION
**Pull request by amanraj-iit**
<hr>

## Description 📜

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 
Fixes Array_and_Equilibrium #8400

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [X] Doc#umentation (Content Creation in the form of codes or tutorials)
- [X] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [X] C/CPP
- [ ] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] JavaScript_#Domain
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [X] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [X] I have performed a self-review of my own code or work.
- [X] I have commented my code, particularly in hard-to-understand areas.
- [X] My changes generates no new warnings.
- [X] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>
[requirements.txt](https://github.com/girlscript/winter-of-contributing/files/7576935/requirements.txt)

### CODE REVIEW

**Comment by amanraj-iit** [comment]
@Saviour1001

**Comment by amanraj-iit** [comment]
@Saviour1001 sir please review

**Comment by vasu-1** [comment]
image
Mention your issue number
remove hash signs from c/cpp and documentation !

File: C_CPP/Competitive Programming/Array_and_Equilibrium_index.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+In other words, the equilibrium index of an array is an index i such that the sum of the elements at indices less than i is equal to the sum of the elements at indices greater than i.
+Element at index i is not included in either part as we know that and it is stated that if more than one equilibrium index is present, you need to return the first one. Also, return -1 if no equilibrium index is present
+
+### Linear-time Solution
+We will solve this problem in linear time by using extra space. 
+The idea is to create an auxiliary array left[], where left[i] stores a sum of elements of the subarray A[0…i-1]. After left[] is filled, traverse the array from right to left and update them right subarray sum for each element encountered. Now, if the sum of elements of the left subarray A[0…i-1] is equal to the sum of elements of the right subarray A[i+1…n) for element A[i], we have found the equilibrium index at i.
+
+### implementation of the above approach in C language
+
+```
```
Comment (vasu-1) [inline C_CPP/Competitive Programming/Array_and_Equilibrium_index.md:24]:
```suggestion
```cpp
```

File: C_CPP/Competitive Programming/Array_and_Equilibrium_index.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
+# An equilibrium index 
+It is an index into the sequence such that the sum of elements at lower indices is equal to the sum of elements at higher indices.
+
+## Statement: 
+For an array A containing n elements, index i is an equilibrium index if the sum of the elements of the subarray A[0…i-1] is equal to the sum of the elements of the subarray A[i+1…n-1] . In other words
+
+(A[0] + A[1] + … + A[i-1]) = (A[i+1] + A[i+2] + … + A[n-1]), where 0 < i < n-1
```
Comment (vasu-1) [inline C_CPP/Competitive Programming/Array_and_Equilibrium_index.md:7]:
```suggestion
`(A[0] + A[1] + … + A[i-1]) = (A[i+1] + A[i+2] + … + A[n-1]), where 0 < i < n-1`
```

**Review by vasu-1 — CHANGES_REQUESTED** [review]
beautify your document with markdown formatting !
Plagiarisms detected !
image

Make it unique (use your own words)

**Comment by amanraj-iit** [comment]
Done , please review

**Comment by vasu-1** [comment]
image
still lots of plagiarised content !
Please make it unique !

**Comment by vasu-1** [comment]
Add your file link into respective readme.md

**Comment by amanraj-iit** [comment]
@vasu-1 please review

File: C_CPP/Competitive Programming/Array_and_Equilibrium_index.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+int right = 0;
+
+//Now we will do traverseing the array from right to left
+for (int i = n - 1; i >= 0; i--)
+just in case if the add of parts of subarray `A[0…i-1]` is adequate
+the add of parts of the subarray `A[i+1…n)` in different words-
+`(A[0] + A[1] + … + A[i-1])` = `(A[i+1] + A[i+2] + … + A[n-1])`
+*/
+
+ if (left[i] == right) {
```
Comment (vasu-1) [inline C_CPP/Competitive Programming/Array_and_Equilibrium_index.md:46]: fix this comment

**Review by vasu-1 — CHANGES_REQUESTED** [review]

**Review by vasu-1 — CHANGES_REQUESTED** [review]
Please add file link first !!!!!!!!!!!!!!!!!!!!!!!!!!

File: C_CPP/Competitive Programming/Array_and_Equilibrium_index.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+// here `right` stores the add of parts of subarray `A[i+1…n)`
+int right = 0;
+
+//Now we will do traverseing the array from right to left
+for (int i = n - 1; i >= 0; i--)
+just in case if the add of parts of subarray `A[0…i-1]` is adequate
+the add of parts of the subarray `A[i+1…n)` in different words-
+`(A[0] + A[1] + … + A[i-1])` = `(A[i+1] + A[i+2] + … + A[n-1])`
+
```
Comment (vasu-1) [inline C_CPP/Competitive Programming/Array_and_Equilibrium_index.md:44]: fix this

**Review by vasu-1 — CHANGES_REQUESTED** [review]

**Comment by amanraj-iit** [comment]
Done @vasu-1 please review

**Comment by vasu-1** [comment]
pahle for loop ke niche jo dikh raha h usko fix karo

**Comment by amanraj-iit** [comment]
@vasu-1  Done

File: C_CPP/Competitive Programming/Array_and_Equilibrium_index.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+for (int i = 1; i < n; i++)
+
+// here `right` stores the add of parts of subarray `A[i+1…n)`
+int right = 0;
+
+//Now we will do traverseing the array from right to left
+for (int i = n - 1; i >= 0; i--)
+{
+       
+        */
```
Comment (vasu-1) [inline C_CPP/Competitive Programming/Array_and_Equilibrium_index.md:43]:
```suggestion
```

Remove this

File: C_CPP/Competitive Programming/Array_and_Equilibrium_index.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+`(A[0] + A[1] + … + A[i-1]) = (A[i+1] + A[i+2] + … + A[n-1]), wherever 0 < i < n-1`
+Here,
+zero is associate degree equilibrium index if A[1] + A[2] + … + A[n-1] sums to zero associate degreed n-1 is an equilibrium index if A[0] + A[1] + … + A[n-2] sums to zero.
+
+
+For example, think about the array . . The equilibrium index is found at index zero, 3, and 7
+
+In different words, the equilibrium index of associate degree array is outlined as associate degree index i that|during which|within which} the add of the weather at indices but i is adequate the add of the weather at indices which is bigger than i.
+Element at index i isn't enclosed in either half as we all know that and it's expressed that if quite one equilibrium index is gift, we'd like to come back the primary one. Also, come back -1 if no equilibrium index is gift
```
Comment (vasu-1) [inline C_CPP/Competitive Programming/Array_and_Equilibrium_index.md:15]: Make it Appropriate !

Then you are good to go for merge !

**Review by vasu-1 — CHANGES_REQUESTED** [review]

**Comment by amanraj-iit** [comment]
@vasu-1 please review

**Review by vasu-1 — CHANGES_REQUESTED** [review]
image
Make it complete
Lots of error in the code snippet and in document

**Comment by amanraj-iit** [comment]
@vasu-1 sir

File: C_CPP/Competitive Programming/Array_and_Equilibrium_index.md (hunk 1 above)
Comment (vasu-1) [reply C_CPP/Competitive Programming/Array_and_Equilibrium_index.md:26]: > 

do this change

**Review by vasu-1 — CHANGES_REQUESTED** [review]

**Review by vasu-1 — CHANGES_REQUESTED** [review]
.

**Review by vasu-1 — APPROVED** [review]
@Suvraneel @Saviour1001 check it

**Comment by Suvraneel** [comment]
@amanraj-iit  LGTM 🚀  Good work :octocat:

**Pull request merged**